src/test/java/uk/gov/ons/ctp/integration/envoycuc/steps/LimitTestSteps.java

```
@@ -0,0 +1,288 @@
+          final int noRequests,
+          final String productGroup,
+          final String deliveryChannel,
+          final String caseType,
+          final String individualStr,
+          final String telephone,
+          final String ipAddress,
+          final String uprnStr) {
+
+    final String fullTelephone = rateLimiterClientprovider.getTestValuePrefix() + telephone;
+    final String fullIpAddress = rateLimiterClientprovider.getTestValuePrefix() + ipAddress;
+    final String fullUprnStr = rateLimiterClientprovider.getTestValuePrefix() + uprnStr;
+
+    for (int i = 0; i < noRequests; i++) {
+      final RateLimiterClientRequest rateLimiterClientRequest =
+          getRateLimiterClientRequest(
+              productGroup,
+              deliveryChannel,
+              caseType,
+              individualStr,
+              fullUprnStr,
+              fullTelephone,
+              fullIpAddress);
+
+      rateLimiterClientRequestContext.getRateLimiterRequestList().add(rateLimiterClientRequest);
+    }
+  }
+
+  @When("I post the fulfilments to the envoy proxy client")
+  public void iPostTheFulfilmentsToTheEnvoyproxyClient() {
+    final TestClient client = getTestClient();
+
+    final List<Boolean> passFailList = rateLimiterClientRequestContext.getPassFail();
+    for (RateLimiterClientRequest r : rateLimiterClientRequestContext.getRateLimiterRequestList()) {
+      boolean isPass = true;
+      try {
+        client.checkRateLimit(
+            r.getDomain(),
+            r.getProduct(),
+            r.getCaseType(),
+            r.getIpAddress(),
+            r.getUprn(),
+            r.getTelNo());
+      } catch (ResponseStatusException ex) {
+        HttpStatus httpStatus = ex.getStatus();
+        if (httpStatus == HttpStatus.TOO_MANY_REQUESTS) {
+          isPass = false;
+          log.info("Rate Limit Exceeded: " + ex.getReason());
+        }
+      } catch (Exception unexpectedException) {
```

why not remove this catch and let it propagate as a failed test?

---

yeah we discussed this - evolution